website/integrations: add incus (#24519)
goauthentik/authentik: website/integrations/hypervisors-orchestrators/incus/index.mdx

---
title: Integrate with Incus
sidebar_label: Incus
support_level: community
---

import RedirectURI20265Note from "../../\_redirect-uri-2026-5-note.mdx";

## What is Incus?

> > Incus is a system container and virtual machine manager that provides a unified experience for running and managing Linux systems.

Reviewer: ```suggestion
> Incus is an open-source manager for Linux system containers, application containers, and virtual machines.
```

>
> -- https://linuxcontainers.org/incus/

## Preparation

The following placeholders are used in this guide:

- `incus.company` is the FQDN of the Incus server.
- `authentik.company` is the FQDN of the authentik installation.

:::warning Incus authorization

> By default, every user who authenticates through the configured OIDC provider has full access to Incus. Before allowing untrusted users to authenticate, configure Incus to authorize OIDC clients with OpenFGA or an authorization scriptlet.

Reviewer 2: Only FGA is supported with oidc

        - Set **Client type** to `Public`.
        - Add a **Redirect URI** of type `Strict` `Authorization` as `https://incus.company/oidc/callback`.
        - Add a **Redirect URI** of type `Strict` `Post Logout` as `https://incus.company`.
        - Select any available signing key.
    - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/bindings-overview/) (policy, group, or user) to manage access to Incus and its listing on a user's **Application Dashboard** page.

3. Click **Submit** to save the new application and provider.

If users access Incus through a port other than `443`, include that port in both redirect URIs. For example, use `https://incus.company:8443/oidc/callback` and `https://incus.company:8443` when connecting directly to the Incus server on port `8443`.

### Create and apply a device code flow

The Incus CLI uses the OAuth device authorization grant to authenticate users. If the active brand already has a device code flow configured, skip this section.

1. Navigate to **Flows and Stages** > **Flows** and click **Create**.
2. Configure the following settings:

>     - **Name**: enter a descriptive name, such as `Incus device code flow`.

Reviewer 2: I dont think this flow should contain incus since its not specific to it
